client/dashboard/src/pages/sources/SourceDetails.tsx

```diff
@@ -0,0 +1,387 @@
+            <div className="flex h-10 w-10 items-center justify-center rounded-lg bg-surface-secondary">
+              <SourceIcon className="h-5 w-5 text-muted-foreground" />
+            </div>
+            <div>
+              <div className="flex items-center gap-2">
+                <h1 className="text-2xl font-semibold">
+                  {source?.name || sourceSlug}
+                </h1>
+                <Badge variant="outline">{sourceType}</Badge>
+              </div>
+              <Type className="text-muted-foreground text-sm" as="p">
+                {source?.slug}
+              </Type>
+            </div>
+          </div>
+          <div className="flex gap-2">
+            <Button
+              variant="secondary"
+              size="sm"
+              icon={<Eye />}
+              onClick={handleViewSpec}
+            >
+              View {isOpenAPI ? "Spec" : "Manifest"}
+            </Button>
+            <Button
+              variant="secondary"
+              size="sm"
+              icon={<Download />}
+              onClick={handleDownload}
+            >
+              Download
+            </Button>
+          </div>
+        </div>
+
+        <div className="space-y-6">
+          {/* Source Metadata Card */}
+          <div className="rounded-lg border bg-card p-6">
+            <Type as="h2" className="text-lg font-semibold mb-4">
+              Source Information
+            </Type>
+            <div className="grid grid-cols-2 gap-4">
+              <div>
+                <Type className="text-sm text-muted-foreground mb-1">Name</Type>
+                <Type className="font-medium">{source?.name}</Type>
+              </div>
+              <div>
+                <Type className="text-sm text-muted-foreground mb-1">Slug</Type>
+                <Type className="font-mono text-sm">{source?.slug}</Type>
+              </div>
+              <div>
+                <Type className="text-sm text-muted-foreground mb-1">Type</Type>
+                <Type className="font-medium">{sourceType}</Type>
+              </div>
+              {!isOpenAPI && source && "runtime" in source && (
+                <div>
+                  <Type className="text-sm text-muted-foreground mb-1">
+                    Runtime
+                  </Type>
+                  <Type className="font-mono text-sm">{source.runtime}</Type>
+                </div>
+              )}
+              <div>
+                <Type className="text-sm text-muted-foreground mb-1">
+                  Last Updated
+                </Type>
+                <div className="flex items-center gap-2">
+                  <Calendar className="h-4 w-4 text-muted-foreground" />
+                  <Type className="text-sm">{lastUpdated}</Type>
+                </div>
+              </div>
+            </div>
+          </div>
+
+          {/* Toolsets Using This Source */}
+          <div className="rounded-lg border bg-card p-6">
```

Comment:
Can we make this its own component in `components/sources/SourceToolsetsCard.tsx`?